<a href="https://colab.research.google.com/github/PhamBaVuong/DataAnalysisDeepLearning/blob/main/Lab06.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
from google.colab import files
import io

In [ ]:
uploaded = files.upload()

Saving 04_CIGARET.csv to 04_CIGARET (2).csv


In [ ]:
data_cigaret = pd.read_csv(io.StringIO(uploaded['04_CIGARET.csv'].decode('utf-8')))

In [ ]:
data_cigaret

KgTar  KgNic  KgCO  MnTar  MnNic  MnCO  FLTar  FLNic  FLCO
0      20    1.1    16     16    1.1    15      5    0.4     4
1      27    1.7    16     13    0.8    17     16    1.0    19
2      27    1.7    16     16    1.0    19     17    1.2    17
3      20    1.1    16      9    0.9     9     13    0.8    18
4      20    1.1    16     14    0.8    17     13    0.8    18
5      24    1.4    17     13    0.8    17     14    1.0    13
6      20    1.1    16     12    0.8    15     15    1.1    17
7      23    1.4    15     14    0.8    17     15    1.1    15
8      20    1.0    16     14    0.9    15     15    1.1    15
9      22    1.2    14     13    0.8    17      9    0.8    12
10     20    1.1    16     13    0.8    17     13    0.8    18
11     20    1.1    16     16    1.2    15     13    0.8    17
12     20    1.1    16     13    0.8    17     13    0.8    18
13     20    1.1    16     13    0.8    17     15    1.0    16
14     20    1.1    16     18    1.3    18      2    0.2     3
15     10    1.8    14      9    0.7    11     15    1.1    18
16     24    1.6    16     19    1.4    18     15    1.0    15
17     20    1.1    16      2    0.2     3     13    0.8    18
18     21    1.2    14     13    0.8    17     14    1.0    15
19     25    1.5    18     14    1.0    14     15    0.9    17
20     23    1.3    15     14    0.8    15     16    1.1    15
21     20    1.1    16     15    0.8    22     15    1.1    15
22     22    1.3    14     16    1.2    16      7    0.6     7
23     20    1.1    16      6    0.6     7     17    1.3    16
24     20    1.1    16      8    0.7     9     15    1.1    14

In [ ]:
import statsmodels.formula.api as smf

# **Bài 1**

In [ ]:
result = smf.ols('KgNic~KgTar+KgCO',data_cigaret).fit()
print(result.summary())

                            OLS Regression Results                            
Dep. Variable:                  KgNic   R-squared:                       0.099
Model:                            OLS   Adj. R-squared:                  0.017
Method:                 Least Squares   F-statistic:                     1.211
Date:                Thu, 28 Oct 2021   Prob (F-statistic):              0.317
Time:                        16:53:22   Log-Likelihood:                 2.7726
No. Observations:                  25   AIC:                            0.4548
Df Residuals:                      22   BIC:                             4.111
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.5937      0.793      2.010      0.0

Ước lượng giá trị KgNic dựa trên 2 thông số KgTar và KgCo là không phù hợp vì R-squared quá bé. P-value > 0.05 nên hệ số hồi quy trên không có ý nghĩa thống kê

# **Bài 2**

In [ ]:
uploaded1 = files.upload()

Saving 23_HOMES.csv to 23_HOMES (2).csv


In [ ]:
data_homes = pd.read_csv(io.StringIO(uploaded1['23_HOMES.csv'].decode('utf-8')))

In [ ]:
data_homes.head(5)

Selling_Price  List_Price  Area  Acres  ...  Taxes  Rooms  Bedrooms  Baths_full
0         400000      414000  2704   2.27  ...   4920      9         3           3
1         370000      379000  2096   0.75  ...   4113      8         4           2
2         382500      389900  2737   1.00  ...   6072      9         4           2
3         300000      299900  1800   0.43  ...   4024      8         4           2
4         305000      319900  1066   3.60  ...   3562      6         3           2

[5 rows x 9 columns]

In [ ]:
result_list_price = smf.ols('Selling_Price~List_Price',data_homes).fit()
print(result_list_price.summary())

                            OLS Regression Results                            
Dep. Variable:          Selling_Price   R-squared:                       0.990
Model:                            OLS   Adj. R-squared:                  0.990
Method:                 Least Squares   F-statistic:                     3697.
Date:                Thu, 28 Oct 2021   Prob (F-statistic):           1.79e-39
Time:                        16:53:33   Log-Likelihood:                -417.67
No. Observations:                  40   AIC:                             839.3
Df Residuals:                      38   BIC:                             842.7
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     99.1841   5792.351      0.017      0.9

In [ ]:
result_area = smf.ols('Selling_Price~Area',data_homes).fit()
print(result_area.summary())

                            OLS Regression Results                            
Dep. Variable:          Selling_Price   R-squared:                       0.643
Model:                            OLS   Adj. R-squared:                  0.633
Method:                 Least Squares   F-statistic:                     68.30
Date:                Thu, 28 Oct 2021   Prob (F-statistic):           5.14e-10
Time:                        16:53:39   Log-Likelihood:                -488.86
No. Observations:                  40   AIC:                             981.7
Df Residuals:                      38   BIC:                             985.1
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   1.339e+05   2.65e+04      5.057      0.0

In [ ]:
result_acres = smf.ols('Selling_Price~Acres',data_homes).fit()
print(result_acres.summary())

                            OLS Regression Results                            
Dep. Variable:          Selling_Price   R-squared:                       0.215
Model:                            OLS   Adj. R-squared:                  0.194
Method:                 Least Squares   F-statistic:                     10.39
Date:                Thu, 28 Oct 2021   Prob (F-statistic):            0.00260
Time:                        16:53:40   Log-Likelihood:                -504.59
No. Observations:                  40   AIC:                             1013.
Df Residuals:                      38   BIC:                             1017.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   3.102e+05   1.55e+04     19.977      0.0

1. Nếu chỉ sử dụng 1 biến x để dự đoán giá nhà, phương trình hồi quy 1 biến dự đoán LP là tốt nhất vì p-value < 0.05 và r-squared có giá trị lớn nhất (gần 1)

In [ ]:
result_list_area = smf.ols('Selling_Price~List_Price+Area',data_homes).fit()
print(result_list_area.summary())

                            OLS Regression Results                            
Dep. Variable:          Selling_Price   R-squared:                       0.990
Model:                            OLS   Adj. R-squared:                  0.989
Method:                 Least Squares   F-statistic:                     1804.
Date:                Thu, 28 Oct 2021   Prob (F-statistic):           1.32e-37
Time:                        16:59:53   Log-Likelihood:                -417.63
No. Observations:                  40   AIC:                             841.3
Df Residuals:                      37   BIC:                             846.3
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    -40.5422   5885.739     -0.007      0.9

In [ ]:
result_list_acres = smf.ols('Selling_Price~List_Price+Acres',data_homes).fit()
print(result_list_acres.summary())

                            OLS Regression Results                            
Dep. Variable:          Selling_Price   R-squared:                       0.990
Model:                            OLS   Adj. R-squared:                  0.989
Method:                 Least Squares   F-statistic:                     1815.
Date:                Thu, 28 Oct 2021   Prob (F-statistic):           1.18e-37
Time:                        17:00:15   Log-Likelihood:                -417.50
No. Observations:                  40   AIC:                             841.0
Df Residuals:                      37   BIC:                             846.1
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   1003.7185   6067.183      0.165      0.8

In [ ]:
result_area_acres = smf.ols('Selling_Price~Area+Acres',data_homes).fit()
print(result_area_acres.summary())

                            OLS Regression Results                            
Dep. Variable:          Selling_Price   R-squared:                       0.815
Model:                            OLS   Adj. R-squared:                  0.805
Method:                 Least Squares   F-statistic:                     81.67
Date:                Thu, 28 Oct 2021   Prob (F-statistic):           2.69e-14
Time:                        17:01:32   Log-Likelihood:                -475.65
No. Observations:                  40   AIC:                             957.3
Df Residuals:                      37   BIC:                             962.4
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   1.113e+05   1.97e+04      5.658      0.0

2. Nếu sử dụng đúng 2 biến dự đoán để dự đoán giá nhà, phương trình hồi quy 2 biến dự đoán LP, LA là tốt nhất vì p-value < 0.05 và r-squared có giá trị lớn


In [ ]:
result_list_area_acres = smf.ols('Selling_Price~List_Price+Area+Acres',data_homes).fit()
print(result_list_area_acres.summary())

                            OLS Regression Results                            
Dep. Variable:          Selling_Price   R-squared:                       0.990
Model:                            OLS   Adj. R-squared:                  0.989
Method:                 Least Squares   F-statistic:                     1178.
Date:                Thu, 28 Oct 2021   Prob (F-statistic):           5.69e-36
Time:                        17:05:16   Log-Likelihood:                -417.50
No. Observations:                  40   AIC:                             843.0
Df Residuals:                      36   BIC:                             849.8
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept   1119.7436   6419.752      0.174      0.8

3. Phương trình hồi quy LP, LA, LOT trong số các phương trình hồi quy trên là tốt nhất để dự đoán giá nhà vì p-value < 0.05, r-squared lớn nhất

**p-value vừa có biến < 0.05 vừa có biến > 0.05 thì phương trình hồi quy đó có ý nghĩa thống kê hay không???**

4. Phương trình hồi quy: 1120 + 0.972 x 400000 + 0.281 x 3000 + 465 x 2 = 391693

Giá trị dự đoán tốt nhất của giá bán là 391693 USD

Giá bán dự đoán có thể là ước lượng tốt vì ít sai lệch nhất so với dữ liệu đầu vào

Giá trị dự đoán này có khả năng rất chính xác